Modules/Devices/src/Devices.Application/Identities/Commands/CancelDeletionProcessAsSupport/Handler.cs

```diff
@@ -28,7 +28,8 @@ public async Task<CancelDeletionAsSupportResponse> Handle(CancelDeletionAsSuppor
 
         await _identitiesRepository.Update(identity, cancellationToken);
 
-        await _notificationSender.SendNotification(identity.Address, new DeletionProcessCancelledBySupportNotification(), cancellationToken);
+        if (identity.Status != IdentityStatus.ToBeDeleted)
+            await _notificationSender.SendNotification(identity.Address, new DeletionProcessCancelledBySupportNotification(), cancellationToken);
```

Comment:
Changes made regarding repetition, and about the notification I think this should not be blocked because in `CancelDeletionProcessAsSupport` the identity goes into status `Active` so it does not make sense to check is it in status `ToBeDeleted`.